services/content/service.go

```diff
@@ -203,7 +203,8 @@ func (s *service) Read(req *api.ReadContentRequest, session api.Content_ReadServ
 		size = oi.Size - offset
 	}
 
-	if offset+size > oi.Size {
+	// it is ok to have `offset+size > oi.Size`. (size=buffer size, oi.Size=blob size)
```

Comment:
How can that read be satisfied? If I ask for 10 bytes and you only provide 5 bytes, we can fulfill the request. The size needs to be correct to 5 bytes. Are you sure there isn't a bug in the `readerAtReader`?

Reply:
Updated the code to let `size = oi.Size - offset` when `offset+size > oi.Size`.

> Are you sure there isn't a bug in the readerAtReader?
Please refer to the below comment